### PR TITLE
Remove the CLI's `GlobalOptions`

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: model/src/main/kotlin/config/OrtConfiguration.kt [hunk 1]
```diff
@@ -67,6 +67,11 @@ data class OrtConfiguration(
      */
     val enableRepositoryPackageCurations: Boolean = false,
 
+    /**
+     * Force overwriting if any existing output files.
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/config/OrtConfiguration.kt:71]: Typo: "if" -> "of"

File: cli/src/main/kotlin/OrtMain.kt [hunk 2]
```diff
@@ -171,8 +168,11 @@ class OrtMain : CliktCommand(name = ORT_NAME, invokeWithoutSubcommand = true) {
         printStackTrace = stacktrace
 
         // Make options available to subcommands and apply static configuration.
-        val ortConfig = OrtConfiguration.load(configArguments, configFile)
-        currentContext.findOrSetObject { GlobalOptions(ortConfig, forceOverwrite) }
+        val ortConfig = OrtConfiguration.load(
+            args = mapOf("ort.forceOverwrite" to "$forceOverwrite") + configArguments,
```
Comment (mnonnenmacher) [inline cli/src/main/kotlin/OrtMain.kt:172]: How does this behave if the option is set in the configuration file? Does it always overwrite (even if the option is not set on the command line) or never?

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: cli/src/main/kotlin/OrtMain.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/OrtMain.kt:172]: Good point, yes, fixed.

Comment (mnonnenmacher) [reply cli/src/main/kotlin/OrtMain.kt]: Should the order not be the other way round, as if I understand that correctly the value from the config file would now always override the command line option?

```kotlin
putAll(configArguments)
if (forceOverwrite) put("ort.forceOverwrite", "true")
```

Comment (sschuberth) [reply cli/src/main/kotlin/OrtMain.kt]: No, `configArguments` only contains the `-P`-style configuration arguments given on the command line, not any configuration from `config.yml` (that's passed in the next line via `configFile`). And as `--force-overwrite` is being deprecated in favor of `-P ort.forceOverwrite=...`, the latter should override the former syntax. That's why I believe the order to be correct.

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**